### ISSUE TITLE
Optional belongs_to association throws NoMethodError 

### ISSUE THREAD

**Issue by rromanchuk** [Netflix/fast_jsonapi#215]
```ruby
class Foo < ApplicationRecord
  belongs_to :bar, optional: true
end
```

```ruby
class FooSerializer
  include FastJsonapi::ObjectSerializer
 
  belongs_to :bar

end
```

On a nil `bar_id` the serializer throws a fatal
`NoMethodError (undefined method `id' for nil:NilClass):`

**Comment by shishirmk** [comment]
This should be released in version 1.2 of the gem

**Comment by kevincolten** [comment]
This does not seem to be fixed in 1.2

**Comment by kevincolten** [comment]
Was able to get it working with https://github.com/Netflix/fast_jsonapi/tree/release-1.2.1#conditional-relationships
